src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/persistence/PersistenceAnnotationQuickFix.java

```diff
@@ -69,15 +70,9 @@ private static void insertAndReplaceAnnotation(Diagnostic diagnostic, JavaCodeAc
         for (PsiAnnotation annotationNode : annotationNodes) {
             ChangeCorrectionProposal proposal = new ModifyAnnotationProposal(name, context.getSource().getCompilationUnit(),
                     context.getASTRoot(), binding, annotationNode, 0, attributes, annotations);
-
-            // Convert the proposal to LSP4J CodeAction
-            // We need to fix all the annotations so all the changes are combined in this one context.
-            // Therefore, we only need to save the last code action.
-            codeAction = context.convertToCodeAction(proposal, diagnostic);
-        }
-        if (codeAction != null) {
-            codeActions.add(codeAction);
         }
```

Comment:
Modified the PersistenceAnnotationQuickFix class to accomodate the changes.

Reply:
Can you describe the behavioural changes you made? It would be good for us to document that.

Reply:
Changes made are : 
A new method `resolveCodeAction` is introduced, responsible for resolving code actions by creating a workspace edit based on the proposed changes.
Instead of creating code actions using a custom method `convertToCodeAction` it now utilizes `JDTUtils.createCodeAction`.
The `getParticipantId` method is overridden to return the class name.